clang/include/clang/Basic/AttrDocs.td

```diff
@@ -2293,6 +2321,33 @@ defaults to ``1``.
 In OpenCL C, this attribute is available in GNU spelling
 (``__attribute__((reqd_work_group_size(X, Y, Z)))``), see section
 6.7.2 Optional Attribute Qualifiers of OpenCL 1.2 specification for details.
+
+.. code-block:: c++
+
+  __kernel __attribute__((reqd_work_group_size(8, 16, 32))) void test() {}
```

Comment:
Is there a reason you specified the __kernel attribute with this spelling? The documentation here says this spelling is for openCL. In SYCL, IIRC this attribute gets propagated to kernel if used in a function called from kernel. I'm not very familiar with openCL but does this not happen for openCL? Can it only be applied to kernel on OpenCL?

Reply:
>>Is there a reason you specified the __kernel attribute with this spelling?

Thanks @elizabethandrews for looking into the patch.

yes, we need to specify __kernel with the attribute (GNU spelling for OpenCL) otherwise we will see the error like this 
"attribute 'reqd_work_group_size' can only be applied to an OpenCL kernel" (refernce: SemaOpenCL/invalid-kernel-attrs.cl file)


>>In SYCL, IIRC this attribute gets propagated to kernel if used in a function called from kernel.

Only SYCL spellings for the attribute 'reqd_work_group_size' such as [[intel::reqd_work_group_size()]]  and [[cl::reqd_work_group_size()]]  gets propagated to kernel if it is used in a function called from SYCL kernel. 
__attribute__((reqd_work_group_size()) is only applied to OpenCL kernel ( reference: SemaOpenCL/invalid-kernel-attrs.cl file).

This is my understanding. @AaronBallman, please correct me if I am wrong.

Reply:
Ok thank you for clarifying!